Library/Homebrew/cli/args.rb

```diff
@@ -12,6 +12,135 @@ def initialize(argv:)
         super
         @argv = argv
       end
+
+      def to_cli_option(name)
+        if name.length == 2
+          "-#{name.tr("?", "")}"
+        else
+          "--#{name.tr("_", "-").tr("?", "")}"
+        end
+      end
+
+      def options_only
+        to_h.keys
+            .map(&:to_s)
+            .reject { |name| %w[argv remaining].include?(name) }
+            .map(&method(:to_cli_option))
+            .select { |arg| arg.start_with?("-") }
+      end
+
+      def flags_only
+        to_h.keys
+            .map(&:to_s)
+            .reject { |name| %w[argv remaining].include?(name) }
+            .map(&method(:to_cli_option))
+            .select { |arg| arg.start_with?("--") }
+      end
+
+      def named
+        remaining
+      end
+
+      def formulae
+        require "formula"
+        # TODO: use @instance variable to ||= cache when moving to CLI::Parser
+        (downcased_unique_named - casks).map do |name|
+          if name.include?("/") || File.exist?(name)
+            Formulary.factory(name, spec)
+          else
+            Formulary.find_with_priority(name, spec)
+          end
+        end.uniq(&:name)
+      end
+
+      def resolved_formulae
+        require "formula"
+        # TODO: use @instance variable to ||= cache when moving to CLI::Parser
+        (downcased_unique_named - casks).map do |name|
+          Formulary.resolve(name, spec: spec(nil))
+        end.uniq(&:name)
+      end
+
+      def casks
+        # TODO: use @instance variable to ||= cache when moving to CLI::Parser
+        downcased_unique_named.grep HOMEBREW_CASK_TAP_CASK_REGEX
```

Comment:
`@casks ||= downcased_unique_named.grep HOMEBREW_CASK_TAP_CASK_REGEX` and remove TODO